### PR TITLE
Align project settings with the other repositories

### PR DESCRIPTION
**Pull request by lcobucci**
Requiring PHP 7.1+, dropping HHVM and simplifying some stuff on the test suite.

### CODE REVIEW

File: lib/Doctrine/DBAL/Migrations/Tools/Console/Command/AbstractCommand.php [hunk 1]
```diff
@@ -117,20 +117,13 @@ protected function getMigrationConfiguration(InputInterface $input, OutputInterf
...
-     * but use the ConfirmationQuestion when available.
-     *
-     * @param $question
+     * @param string $question
      * @param InputInterface $input
      * @param OutputInterface $output
      * @return mixed
      */
-    protected function askConfirmation($question, InputInterface $input, OutputInterface $output)
+    protected function askConfirmation(string $question, InputInterface $input, OutputInterface $output)
```
Comment (alcaeus) [inline lib/Doctrine/DBAL/Migrations/Tools/Console/Command/AbstractCommand.php]: Technically, this is a BC break - people might have overridden this method to add their own logic.

Comment (lcobucci) [inline lib/Doctrine/DBAL/Migrations/Tools/Console/Command/AbstractCommand.php]: Indeed, will remove the type declaration.

File: lib/Doctrine/DBAL/Migrations/Migration.php [hunk 2]
```diff
@@ -95,14 +95,18 @@ public function writeSqlFile($path, $to = null)
 
         $this->outputWriter->write(sprintf("-- Migrating from %s to %s\n", $from, $to));
 
-        $sqlWriter = new SqlFileWriter(
+        return $this->createSqlFileWriter($path)
+                    ->write($sql, $direction);
+    }
+
+    protected function createSqlFileWriter(string $path): SqlFileWriter
```
Comment (Ocramius) [inline lib/Doctrine/DBAL/Migrations/Migration.php]: `private` plz

File: lib/Doctrine/DBAL/Migrations/Version.php [hunk 3]
```diff
@@ -228,14 +228,19 @@ public function writeSqlFile($path, $direction = self::DIRECTION_UP)
         $this->outputWriter->write("\n-- Version " . $this->version . "\n");
 
         $sqlQueries = [$this->version => $queries];
-        $sqlWriter = new SqlFileWriter(
+
+        return $this->createSqlFileWriter($path)
+                    ->write($sqlQueries, $direction);
+    }
+
+    protected function createSqlFileWriter(string $path): SqlFileWriter
```
Comment (Ocramius) [inline lib/Doctrine/DBAL/Migrations/Version.php]: Unless designed for inheritance, `private`

File: composer.json [hunk 4]
```diff
@@ -19,10 +19,8 @@
     },
     "require-dev": {
         "doctrine/orm": "~2.5",
-        "phpunit/phpunit": "~4.7",
+        "phpunit/phpunit": "~6.2",
         "doctrine/coding-standard": "dev-master",
-        "mockery/mockery": "^0.9.4",
-        "johnkary/phpunit-speedtrap": "~1.0@dev",
```
Comment (mikeSimonson) [inline composer.json]: @lcobucci Why are you removing speedtrap ?

File: lib/Doctrine/DBAL/Migrations/Tools/Console/Command/AbstractCommand.php [hunk 5]
```diff
@@ -117,20 +117,13 @@ protected function getMigrationConfiguration(InputInterface $input, OutputInterf
...
      * @param InputInterface $input
      * @param OutputInterface $output
      * @return mixed
      */
     protected function askConfirmation($question, InputInterface $input, OutputInterface $output)
     {
-        if (!$this->getHelperSet()->has('question')) {
-            return $this->getHelper('dialog')->askConfirmation($output, '<question>' . $question . '</question>', false);
-        }
-
```
Comment (mikeSimonson) [inline lib/Doctrine/DBAL/Migrations/Tools/Console/Command/AbstractCommand.php]: There are tests that are tied to this BC feature that you can remove too.

File: .travis.yml [hunk 6]
```diff
@@ -42,5 +42,4 @@ before_install:
   - if [[ ! $GITHUB_TOKEN ]]; then echo "no github token"; fi
 
 after_script:
-  - if [[ $TRAVIS_PHP_VERSION = '5.6' ]]; then php vendor/bin/coveralls -v ; fi
```
Comment (mikeSimonson) [inline .travis.yml]: I am still using it. Is it a problem to keep it ?

File: .travis.yml [hunk 7]
```diff
@@ -1,46 +1,41 @@
...
+dist: trusty
 sudo: false
-
 language: php
 
 php:
-  - 5.6
-  - 7.0
   - 7.1
-  - nigthly
```
Comment (mikeSimonson) [inline .travis.yml]: I would like to keep nightly in there with allowed failures.

File: .travis.yml [hunk 8]
```diff
@@ -1,46 +1,41 @@
...
-  - hhvm
-
-matrix:
-  fast_finish: true
-  include:
-    - php: 5.6
-      env: deps=low
```
Comment (mikeSimonson) [inline .travis.yml]: We also need to keep a 7.1 run with deps low in the matrix.

Comment (lcobucci) [inline .travis.yml]: It's here: https://github.com/doctrine/migrations/pull/537/files#diff-354f30a63fb0907d4ad57269548329e3R26

Comment (mikeSimonson) [inline .travis.yml]: ok

File: lib/Doctrine/DBAL/Migrations/Migration.php (hunk 2 above)
Comment (lcobucci) [inline lib/Doctrine/DBAL/Migrations/Migration.php]: I can't... and it really hurts me.
We were using mockery a big nasty black magic to overload the instantiation of that class. I didn't wan't to do a huge design change to make it work properly with only PHPUnit mocks. 

Suggestion?

File: lib/Doctrine/DBAL/Migrations/Version.php (hunk 3 above)
Comment (lcobucci) [inline lib/Doctrine/DBAL/Migrations/Version.php]: Same reason as https://github.com/doctrine/migrations/pull/537#discussion_r130853713

File: lib/Doctrine/DBAL/Migrations/Tools/Console/Command/AbstractCommand.php (hunk 5 above)
Comment (lcobucci) [inline lib/Doctrine/DBAL/Migrations/Tools/Console/Command/AbstractCommand.php]: Our coverage report said we weren't covering this flow, can you point the test to me?

File: lib/Doctrine/DBAL/Migrations/Migration.php (hunk 2 above)
Comment (Ocramius) [inline lib/Doctrine/DBAL/Migrations/Migration.php]: Got a link to the test? :-\

File: .travis.yml (hunk 6 above)
Comment (lcobucci) [inline .travis.yml]: It's not a big problem to keep it, I just saw on https://coveralls.io/github/doctrine/migrations that the last build was March 2015 so I assumed it was useless here as well

File: composer.json (hunk 4 above)
Comment (lcobucci) [inline composer.json]: Because the stable version doesn't support PHPUnit 6. Also I don't see much value on it right now, the only reason we had slow tests on the suite was the fact we were using mockery overload instance feature and that require tests to run in isolated processes, which has been removed.

I can add it back though 😄

File: lib/Doctrine/DBAL/Migrations/Migration.php (hunk 2 above)
Comment (lcobucci) [inline lib/Doctrine/DBAL/Migrations/Migration.php]: https://github.com/doctrine/migrations/pull/537/files#diff-5a519f363afe6bb6f788510621f527f0L147

Comment (lcobucci) [inline lib/Doctrine/DBAL/Migrations/Migration.php]: Which now is https://github.com/doctrine/migrations/pull/537/files#diff-5a519f363afe6bb6f788510621f527f0R188

Comment (lcobucci) [inline lib/Doctrine/DBAL/Migrations/Migration.php]: And allow us to remove this: https://github.com/doctrine/migrations/pull/537/files#diff-5a519f363afe6bb6f788510621f527f0L53

File: lib/Doctrine/DBAL/Migrations/Tools/Console/Command/AbstractCommand.php (hunk 5 above)
Comment (chrisguitarguy) [inline lib/Doctrine/DBAL/Migrations/Tools/Console/Command/AbstractCommand.php]: -
 https://github.com/doctrine/migrations/blob/b82d683019fea695119b1972b07be77378186aff/tests/Doctrine/DBAL/Migrations/Tests/Tools/Console/Command/AbstractCommandTest.php#L309-L336
- https://github.com/doctrine/migrations/blob/b82d683019fea695119b1972b07be77378186aff/tests/Doctrine/DBAL/Migrations/Tests/Tools/Console/Command/DialogSupport.php

File: .travis.yml (hunk 7 above)
Comment (Majkl578) [inline .travis.yml]: That error seems unrelated, looks like you force-pushed the branch before Travis was able to grab the commit. :P

> the pipeline didn't work even though that one as allowed to fail

Maybe there's a difference betwen `errored` and `failed` state.

File: .travis.yml (hunk 6 above)
Comment (lcobucci) [inline .travis.yml]: Are you sure we're using for doctrine/migrations? Do you want me to put it back?

File: lib/Doctrine/DBAL/Migrations/Tools/Console/Command/AbstractCommand.php (hunk 5 above)
Comment (lcobucci) [inline lib/Doctrine/DBAL/Migrations/Tools/Console/Command/AbstractCommand.php]: Done 😉

File: .travis.yml (hunk 7 above)
Comment (lcobucci) [inline .travis.yml]: I added it back, but I had a different error before. Let's see if it happens again.

Comment (lcobucci) [inline .travis.yml]: Now it works... I'm probably getting crazier 😜

File: lib/Doctrine/DBAL/Migrations/Migration.php (hunk 2 above)
Comment (Ocramius) [inline lib/Doctrine/DBAL/Migrations/Migration.php]: @lcobucci urgh. I am really reluctant to introduce a new `protected` method that will be abused till eternity :S

A huge test change would be more acceptable than this, IMO...

Comment (lcobucci) [inline lib/Doctrine/DBAL/Migrations/Migration.php]: Ok, I'll go down the rabbit hole on this one. However it would probably not be only a test change.

**Comment by lcobucci** [comment]
@Ocramius @mikeSimonson I think this is good to go, please take a look 😉

File: phpunit.xml.dist [hunk 9]
```diff
@@ -12,26 +17,4 @@
...
-            </arguments>
-        </listener>
-    </listeners>
-
-    <logging>
-        <log type="coverage-html" target="build/coverage" title="doctrine migrations"
-             charset="UTF-8" yui="true" highlight="true" lowUpperBound="5"
-             highLowerBound="70" />
-        <log type="coverage-clover" target="build/logs/clover.xml"/>
-    </logging>
```
Comment (mikeSimonson) [inline phpunit.xml.dist]: Is there any reason to remove those two configurations ?
They are quite handy to work locally.

The rest of the PR is GTG for me.

Comment (alcaeus) [inline phpunit.xml.dist]: @mikeSimonson you can always create your customized `phpunit.xml` locally: it overrides the `.dist` file and allows you to have custom settings.

File: lib/Doctrine/DBAL/Migrations/FileQueryWriter.php [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+
+    public function __construct(string $columnName, string $tableName, ?OutputWriter $outputWriter)
+    {
+        $this->columnName   = $columnName;
+        $this->tableName    = $tableName;
+        $this->outputWriter = $outputWriter;
+    }
+
+    /**
+     * TODO: move SqlFileWriter's behaviour to this class - and kill it with fire (on the next major release)
```
Comment (Ocramius) [inline lib/Doctrine/DBAL/Migrations/FileQueryWriter.php]: TODO? :-P

File: lib/Doctrine/DBAL/Migrations/FileQueryWriter.php [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+            $this->columnName,
+            $this->tableName,
+            $path,
+            $this->outputWriter
+        );
+
+        $result = $writer->write($queriesByVersion, $direction);
+
+        // SqlFileWriter#write() returns `bool|int` but all clients expect it to be `bool` only
+        return $result !== false && $result > 0;
```
Comment (Ocramius) [inline lib/Doctrine/DBAL/Migrations/FileQueryWriter.php]: Isn't a `(bool)` cast sufficient?

File: lib/Doctrine/DBAL/Migrations/FileQueryWriter.php [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+        $this->tableName    = $tableName;
+        $this->outputWriter = $outputWriter;
+    }
+
+    /**
+     * TODO: move SqlFileWriter's behaviour to this class - and kill it with fire (on the next major release)
+     */
+    public function write(string $path, string $direction, array $queriesByVersion) : bool
+    {
+        $writer = new SqlFileWriter(
```
Comment (Ocramius) [inline lib/Doctrine/DBAL/Migrations/FileQueryWriter.php]: What about the injected writer?

Comment (lcobucci) [inline lib/Doctrine/DBAL/Migrations/FileQueryWriter.php]: I don't get what you mean, this is the implementation of the injected writer. I just don't wanted to implement everything here at this moment, it feels way too far from the scope of the PR.

File: phpunit.xml.dist (hunk 9 above)
Comment (lcobucci) [inline phpunit.xml.dist]: Exactly 😁

File: lib/Doctrine/DBAL/Migrations/FileQueryWriter.php (hunk 11 above)
Comment (lcobucci) [inline lib/Doctrine/DBAL/Migrations/FileQueryWriter.php]: Yes it is, silly me...

File: lib/Doctrine/DBAL/Migrations/FileQueryWriter.php (hunk 10 above)
Comment (Ocramius) [inline lib/Doctrine/DBAL/Migrations/FileQueryWriter.php]: Can you spawn an issue for this TODO?

File: lib/Doctrine/DBAL/Migrations/FileQueryWriter.php (hunk 12 above)
Comment (Ocramius) [inline lib/Doctrine/DBAL/Migrations/FileQueryWriter.php]: Ah, missed that you are actually using `$this->outputWriter` :man_facepalming:

File: lib/Doctrine/DBAL/Migrations/FileQueryWriter.php (hunk 10 above)
Comment (lcobucci) [inline lib/Doctrine/DBAL/Migrations/FileQueryWriter.php]: I was just waiting to get it merged to open the issueS 😜

**Comment by Ocramius** [comment]
:ship:
